plotly/tests/test_optional/test_tools/test_figure_factory.py

```diff
@@ -798,64 +834,70 @@ def test_annotated_heatmap_kwargs(self):
                                                                    [1,
                                                                     '#e6005a']]
                                                        )
-        expected_a = {'data': [{'colorscale': [[0, '#ffffff'], [1, '#e6005a']],
+        expected_a = {'data': [{'colorscale': ((0, '#ffffff'), (1, '#e6005a')),
```

Comment:
`colorscales` are still accepted as nested lists, but they are coerced into nested tuples during validation.  This is so that they are immutable.  If I left them as lists, people could do things like
```
fig.data[0].marker.colorscale[0][1] = 'green'
```
and then I would know to update the frontend.